Build fails with `error: ‘find_if’ is not a member of ‘std’; did you mean ‘find’?`

### Troubleshooting docs

- [X] My problem is not solved in the Troubleshooting docs

### Anaconda default channels

- [X] I do NOT use the Anaconda default channels (pkgs/* etc.)

### How did you install Mamba?

Other (please describe)

### Search tried in issue tracker

find_if

### Latest version of Mamba

- [X] My problem is not solved with the latest version

### Tried in Conda?

I have this problem with Conda as well, without using Mamba

### Describe your issue

Trying to build commit 16fec215e9de533b0156ff262aa65530c3212408 on Fedora Rawhide:
```
[ 50%] Building CXX object libmamba/CMakeFiles/libmamba-dyn.dir/src/download/mirror_map.cpp.o
cd /home/orion/fedora/libmamba/libmamba-2.0.0-build/mamba-16fec215e9de533b0156ff262aa65530c3212408/redhat-linux-build/libmamba && /usr/bin/g++ -DFMT_SHARED -DLIBMAMBA_EXPORTS -DSIMDJSON_THREADS_ENABLED=1 -DSPDLOG_ACTIVE_LEVEL=SPDLOG_LEVEL_TRACE -DSPDLOG_FMT_EXTERNAL -Dlibmamba_dyn_EXPORTS -I/home/orion/fedora/libmamba/libmamba-2.0.0-build/mamba-16fec215e9de533b0156ff262aa65530c3212408/libmamba/include -I/home/orion/fedora/libmamba/libmamba-2.0.0-build/mamba-16fec215e9de533b0156ff262aa65530c3212408/libmamba/src -I/home/orion/fedora/libmamba/libmamba-2.0.0-build/mamba-16fec215e9de533b0156ff262aa65530c3212408/libmamba/ext/solv-cpp/include -isystem /usr/share -O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -m64 -march=x86-64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -mtls-dialect=gnu2 -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -O2 -g -DNDEBUG -fPIC -Wall -Wextra -Wshadow -Wnon-virtual-dtor -Wold-style-cast -Wcast-align -Wunused -Woverloaded-virtual -Wpedantic -Wconversion -Wsign-conversion -Wnull-dereference -Wdouble-promotion -Wformat=2 -Wunreachable-code -Wuninitialized -Wmisleading-indentation -Wduplicated-cond -Wduplicated-branches -Wlogical-op -Wuseless-cast -MD -MT libmamba/CMakeFiles/libmamba-dyn.dir/src/download/mirror_map.cpp.o -MF CMakeFiles/libmamba-dyn.dir/src/download/mirror_map.cpp.o.d -o CMakeFiles/libmamba-dyn.dir/src/download/mirror_map.cpp.o -c /home/orion/fedora/libmamba/libmamba-2.0.0-build/mamba-16fec215e9de533b0156ff262aa65530c3212408/libmamba/src/download/mirror_map.cpp
In file included from /home/orion/fedora/libmamba/libmamba-2.0.0-build/mamba-16fec215e9de533b0156ff262aa65530c3212408/libmamba/src/download/mirror_map.cpp:7:
/home/orion/fedora/libmamba/libmamba-2.0.0-build/mamba-16fec215e9de533b0156ff262aa65530c3212408/libmamba/include/mamba/download/mirror_map.hpp: In member function ‘MirrorType& mamba::download::mirror_map::create_unique_mirror(const std::string&, Args&& ...)’:
/home/orion/fedora/libmamba/libmamba-2.0.0-build/mamba-16fec215e9de533b0156ff262aa65530c3212408/libmamba/include/mamba/download/mirror_map.hpp:74:26: error: ‘find_if’ is not a member of ‘std’; did you mean ‘find’?
   74 |         auto iter = std::find_if(
      |                          ^~~~~~~
      |                          find
/home/orion/fedora/libmamba/libmamba-2.0.0-build/mamba-16fec215e9de533b0156ff262aa65530c3212408/libmamba/src/download/mirror_map.cpp: In function ‘bool mamba::download::{anonymous}::contains(It, It, mamba::download::MirrorID)’:
/home/orion/fedora/libmamba/libmamba-2.0.0-build/mamba-16fec215e9de533b0156ff262aa65530c3212408/libmamba/src/download/mirror_map.cpp:41:25: error: ‘find_if’ is not a member of ‘std’; did you mean ‘find’?
   41 |             return std::find_if(first, last, [id](const auto& mirror) { return id == mirror->id(); })
      |                         ^~~~~~~
      |                         find
[ 51%] Building CXX object libmamba/CMakeFiles/libmamba-dyn.dir/src/download/mirror_impl.cpp.o
```

gcc-14.1.1-7.fc41.x86_64

### mamba info / micromamba info

_No response_

### Logs

_No response_

### environment.yml

_No response_

### ~/.condarc

_No response_

--- a/libmamba/include/mamba/download/mirror_map.hpp
+++ b/libmamba/include/mamba/download/mirror_map.hpp
@@ -7,6 +7,7 @@
 #ifndef MAMBA_DOWNLOAD_MIRROR_MAP_HPP
 #define MAMBA_DOWNLOAD_MIRROR_MAP_HPP
 
+#include <algorithm>
 #include <memory>
 #include <string>
 #include <string_view>

